src/components/CardEditor/CardEditForm/CardEditFormItems/DataSeriesFormItems/DataSeriesFormContent.jsx

```diff
@@ -377,54 +451,89 @@ const DataSeriesFormItem = ({
           id: dataItem.dataSourceId,
           content: {
             value: dataItem.label,
-            icon: isComplexDataSeries ? (
-              <div
-                style={{
-                  width: '1rem',
-                  height: '1rem',
-                  backgroundColor:
-                    dataItem.color ||
-                    DATAITEM_COLORS_OPTIONS[i % DATAITEM_COLORS_OPTIONS.length],
-                }}
-              />
-            ) : null,
+            icon:
+              cardConfig.type === CARD_TYPES.TIMESERIES ||
+              cardConfig.type === CARD_TYPES.BAR ? (
+                <div
+                  style={{
```

Comment:
Do we have to inline these styles? We should add classes and update custom css properties